controllers/nonadmin_controller.go

```diff
@@ -0,0 +1,193 @@
+			ensureRequiredSpecs(nonAdminDeployment, nonAdminImage)
+
+			// Setting controller owner reference on the non admin controller deployment
+			return controllerutil.SetControllerReference(&dpa, nonAdminDeployment, r.Scheme)
+		},
+	)
+	if err != nil {
+		return false, err
+	}
+
+	if operation != controllerutil.OperationResultNone {
+		r.EventRecorder.Event(
+			nonAdminDeployment,
+			corev1.EventTypeNormal,
+			"NonAdminDeploymentReconciled",
+			fmt.Sprintf("Non admin controller deployment %s/%s %s", nonAdminDeployment.Namespace, nonAdminDeployment.Name, operation),
+		)
+	}
+	return true, nil
+}
+
+func ensureRequiredLabels(deploymentObject *appsv1.Deployment) {
+	maps.Copy(deploymentLabels, controlPlaneLabel)
+	deploymentObjectLabels := deploymentObject.GetLabels()
+	if deploymentObjectLabels == nil {
+		deploymentObject.SetLabels(deploymentLabels)
+	} else {
+		for key, value := range deploymentLabels {
+			deploymentObjectLabels[key] = value
+		}
+		deploymentObject.SetLabels(deploymentObjectLabels)
+	}
+}
+
+func ensureRequiredSpecs(deploymentObject *appsv1.Deployment, image string) {
+	deploymentObject.Spec.Replicas = pointer.Int32(1)
+	deploymentObject.Spec.Selector = &metav1.LabelSelector{
+		MatchLabels: controlPlaneLabel,
+	}
+	templateObjectLabels := deploymentObject.Spec.Template.GetLabels()
+	if templateObjectLabels == nil {
+		deploymentObject.Spec.Template.SetLabels(controlPlaneLabel)
+	} else {
+		templateObjectLabels[controlPlaneKey] = controlPlaneLabel[controlPlaneKey]
+		deploymentObject.Spec.Template.SetLabels(templateObjectLabels)
+	}
+	if len(deploymentObject.Spec.Template.Spec.Containers) == 0 {
+		deploymentObject.Spec.Template.Spec.Containers = []corev1.Container{{
+			Name:            containerName,
+			Image:           image,
+			ImagePullPolicy: corev1.PullAlways,
+		}}
+	} else {
+		for _, container := range deploymentObject.Spec.Template.Spec.Containers {
+			if container.Name == containerName {
+				container.Image = image
+				container.ImagePullPolicy = corev1.PullAlways
+				break
+			}
+		}
+	}
+	deploymentObject.Spec.Template.Spec.RestartPolicy = corev1.RestartPolicyAlways
+	deploymentObject.Spec.Template.Spec.ServiceAccountName = nonAdminControllerManager
+}
+
+func (r *DPAReconciler) checkNonAdminEnabled(dpa *oadpv1alpha1.DataProtectionApplication) bool {
+	// TODO https://github.com/openshift/oadp-operator/pull/1316
+	if dpa.Spec.Features != nil && dpa.Spec.Features.EnableNonAdmin != nil {
+		return *dpa.Spec.Features.EnableNonAdmin
+	}
+
+	return false
+}
+
+func (r *DPAReconciler) getNonAdminImage(dpa *oadpv1alpha1.DataProtectionApplication) string {
+	// TODO https://github.com/openshift/oadp-operator/pull/1316
+	unsupportedOverride := dpa.Spec.UnsupportedOverrides[oadpv1alpha1.NonAdminControllerImageKey]
+	if unsupportedOverride != "" {
+		return unsupportedOverride
+	}
+
+	environmentVariable := os.Getenv("RELATED_IMAGE_NON_ADMIN_CONTROLLER")
+	return environmentVariable
```

Comment:
here I did not add a fallback like it was in previous controller. I think is duplication of information and not needed

do you remember any use cases where OADP operator deployment pod would have its environment variables overwrite or delete?

Reply:
For disconnected env/air gapped nev clusters, `RELATED_IMAGE_NON_ADMIN_CONTROLLER` value on CSV is used

Reply:
I think my question was misunderstood, I want to know if there is any use case of OADP that `RELATED_IMAGE_...` environment variables will NOT be used 

As asked in slack, we change these environment variables for production, but not in here in the code https://github.com/openshift/oadp-operator/blob/fe0444c847f79a8a12d03f3d67208c552af12457/pkg/common/common.go#L60-L69

So, fallback is wrong for production (will open an issue)

Can I have my implementation as it is, i.e., it tries unsupportedOverride, then environment variable and then fails if both are empty?

Reply:
issue https://github.com/openshift/oadp-operator/issues/1379